File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        L = 0
        res = cost = 0

        for R in range(len(s)):
            cost += abs(ord(s[R]) - ord(t[R]))
            if cost > maxCost:
                cost -= abs(ord(s[L]) - ord(t[L]))
                L += 1
            res = max(res, R - L + 1)
        return res

File: 1321-get-equal-substrings-within-budget/get-equal-substrings-within-budget.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        int res = 0, curr = 0, cost = 0;
        int n = s.size();

        for (int i = 0, j = 0; j < n; ++j) {
            cost = abs(s[j] - t[j]);
            curr += cost;

            while (curr > maxCost) {
                curr -= abs(s[i] - t[i]);
                ++i;
            }

            res = max(res, j - i + 1);
        }
        return res;
    }
};